### PR TITLE
[HLS-2907] Mongo Proxy library

### PR DESCRIPTION
**Pull request by lior-govrin**
@nozik, @LironKS, anything else I need to do to expose the new proxy? Add it to a `main` or something like that?

### CODE REVIEW

File: proxy-libs/heliosmongo/mongo.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+var ErrNilCursor = originalMongo.ErrNilCursor
+var ErrNilDocument = originalMongo.ErrNilDocument
+var ErrNilValue = originalMongo.ErrNilValue
+var ErrNoDocuments = originalMongo.ErrNoDocuments
+var ErrNonStringIndexName = originalMongo.ErrNonStringIndexName
+var ErrUnacknowledgedWrite = originalMongo.ErrUnacknowledgedWrite
+var ErrWrongClient = originalMongo.ErrWrongClient
+
+func setMonitor(opts []*options.ClientOptions) []*options.ClientOptions {
+	monitor := otelmongo.NewMonitor()
```
Comment (nozik) [inline proxy-libs/heliosmongo/mongo.go:79]: We add it to each 'opt' - is that necessary?

Comment (nozik) [reply proxy-libs/heliosmongo/mongo.go:79]: In this example - https://github.com/open-telemetry/opentelemetry-go-contrib/blob/9fd4fa889a38e38f73c10d4872bf4a64e275f157/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/example_test.go#L32 - only one client has it.

File: proxy-libs/heliosmongo/mongo_test.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	clientOptions := getClientOptions()
+	client, error := Connect(context.Background(), clientOptions)
+
+	if error != nil {
+		panic(error)
+	}
+
+	insertUser(client, 12345, "Lior Govrin", "Software Engineer")
+	attributes := assertSpan(t, spanRecorder)
+	assertAttributes(t, attributes, 12345, "Lior Govrin", "Software Engineer")
```
Comment (nozik) [inline proxy-libs/heliosmongo/mongo_test.go:96]: Senior Principle Software Engineer

File: proxy-libs/heliosmongo/mongo.go (hunk 1 above)
Comment (lior-govrin) [reply proxy-libs/heliosmongo/mongo.go:79]: Yes, but notice that `Connect` and `NewClient` receive an array of options (`...opts`), so the `Monitor` has to be in all of them (the library merges all the options in the array, so to be sure that the `Monitor` is set, I push it to all the opts.

File: proxy-libs/heliosmongo/mongo_test.go (hunk 2 above)
Comment (lior-govrin) [reply proxy-libs/heliosmongo/mongo_test.go:96]: LOL 😬

**Review by nozik — APPROVED** [review]

**Pull request merged**